datastream-common/src/main/java/com/linkedin/datastream/metrics/DynamicMetricsManager.java

```diff
@@ -161,7 +166,7 @@ private void updateCache(String simpleClassName, String key, String metricName,
     String fullMetricName = MetricRegistry.name(simpleName, key, metricName);
 
     Metric metric = getMetric(fullMetricName, metricClass);
-
+    _registeredMetricRefCount.compute(fullMetricName, (localKey, val) -> (val == null) ? 1 : val + 1);
```

Comment:
Nitpick: Can we move this to line 167?

Reply:
the reason to keep it here, because getMetric can throw exception if there is a name conflict with other metric type.